### PR TITLE
Set the workspace root when doing nft scan

### PR DESCRIPTION
**Pull request by matthewp**
- Previously we were setting the nft scan root to the `/` folder, which causes chaos if you have routes like `/home` or `/dev`. This fixes it by setting it to the workspace root instead.

### CODE REVIEW

**Comment by matthewp** [comment]
!preview netlify-bug

**Comment by matthewp** [comment]
!preview netlify-bug

**Comment by matthewp** [comment]
!preview netlify-bug

**Comment by matthewp** [comment]
!preview netlify-bug

**Comment by matthewp** [comment]
!preview netlify-bug

**Comment by matthewp** [comment]
!preview netlify-bug

**Comment by matthewp** [comment]
!preview netlify-bug

**Comment by matthewp** [comment]
!preview netlify-bug

**Comment by matthewp** [comment]
!preview netlify-bug

**Comment by matthewp** [comment]
!preview netlify-bug

**Comment by matthewp** [comment]
!preview netlify-bug

**Comment by matthewp** [comment]
!preview netlify-bug

**Comment by wobsoriano** [comment]
Tried the snapshot version and I can confirm it is working:

Screenshot 2024-09-07 at 10 25 42 AM

File: packages/netlify/src/lib/nft.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import { posix, relative, sep } from 'node:path';
-import { fileURLToPath } from 'node:url';
+import { fileURLToPath, pathToFileURL } from 'node:url';
 import { copyFilesToFolder } from '@astrojs/internal-helpers/fs';
+import { appendForwardSlash } from '@astrojs/internal-helpers/path';
 import type { AstroIntegrationLogger } from 'astro';
+import { searchForWorkspaceRoot } from 'vite';
```
Comment (ascorbic) [inline packages/netlify/src/lib/nft.ts:6]: today I learned

File: packages/netlify/package.json [hunk 2]
```diff
@@ -35,7 +35,8 @@
     "@astrojs/underscore-redirects": "^0.3.4",
     "@netlify/functions": "^2.8.0",
     "@vercel/nft": "^0.27.4",
-    "esbuild": "^0.21.5"
+    "esbuild": "^0.21.5",
+    "vite": "^5.4.2"
```
Comment (bluwy) [inline packages/netlify/package.json:39]: we no longer need `vite` in deps since we vendor the search code

Comment (alexanderniebuhr) [reply packages/netlify/package.json:39]: Not quite, the PR includes https://github.com/withastro/adapters/pull/381/files#diff-a2534d55417999ca927cbdb98267b817468cec23ba870f9f70a7077710d797bbR6

File: packages/vercel/src/lib/nft.ts
Comment (alexanderniebuhr) [inline packages/vercel/src/lib/nft.ts:1]: Why do we have copied the function for vercel but not for netlify?

**Review by alexanderniebuhr — APPROVED** [review]
Added some thoughts and comments, otherwise lgtm.

File: packages/netlify/package.json (hunk 2 above)
Comment (bluwy) [reply packages/netlify/package.json:39]: Ah right, I thought the vendor happened in both packages

File: packages/vercel/src/lib/nft.ts
Comment (matthewp) [reply packages/vercel/src/lib/nft.ts:1]: For some reason simply having a `import 'vite'` in the Vercel adapter causes tests to timeout. I spent too much time and couldn't figure out why.

Comment (ascorbic) [reply packages/vercel/src/lib/nft.ts:1]: We could move it into our shared utils, like we already did with some other vite stuff

**Pull request merged**